File: DatingApp.API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DatingApp.API.Helpers
{
    public class UserParams :PaginationParams
    {
        public string CurrentuserName { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public string OrderBy { get; set; } = "lastActive";
    }
}

File: clientapp/src/app/_modals/userparams.ts
import { User } from "./user";

export class UserParams {
    gender: string;
    minAge:Number =18;
    maxAge:Number = 99;
    pageNumber:Number =1;
    pageSize:Number = 5;
    orderBy: string = 'lastActive';

    constructor(user:User){
        this.gender = user.gender === "male"? "female":"male";
    }
}